groups/bsl/bslstl/bslstl_optional.t.cpp

```diff
@@ -0,0 +1,1032 @@
+class Test_Util {
+    // This class provided test utilities that have different behaviour
+    // depending on whether 'TYPE is allocator aware or not.  The main template
+    // is for allocator aware types.
+
+  public:
+    static bool checkAllocator(const TYPE&                 obj,
+                               const bsl::allocator<char>& expected);
+        // Check if, for the specified 'obj', 'obj.get_allocator()' returns the
+        // specified 'expected' allocator.
+
+    static bool hasSameAllocator(const TYPE& obj, const TYPE& other);
+        // Check if, for the specified 'obj' and specified 'other',
+        // 'obj.get_allocator() == other.get_allocator()';
+};
+
+template <class TYPE>
+class Test_Util<TYPE, false> {
+    // This class provided test utilities that have different behaviour
+    // depending on whether 'TYPE is allocator aware or not.  This
+    // specialization is for non allocator aware types.
+
+  public:
+    static bool checkAllocator(const TYPE&, const bsl::allocator<char>&);
+        // return 'true'.
+
+    static bool hasSameAllocator(const TYPE&, const TYPE&);
+        // return 'true'.
+};
+
+template <class TYPE, bool USES_BSLMA_ALLOC>
+inline
+bool Test_Util<TYPE, USES_BSLMA_ALLOC>::checkAllocator(
+                                          const TYPE&                 obj,
+                                          const bsl::allocator<char>& expected)
+{
+    return (expected == obj.get_allocator());
+}
+
+template <class TYPE, bool USES_BSLMA_ALLOC>
+inline
+bool Test_Util<TYPE, USES_BSLMA_ALLOC>::hasSameAllocator(const TYPE& obj,
+                                                         const TYPE& other)
+{
+    return (obj.get_allocator() == other.get_allocator());
+}
+
+template <class TYPE>
+inline bool
+Test_Util<TYPE, false>::checkAllocator(const TYPE&,
+                                       const bsl::allocator<char>&)
+{
+    return true;
+}
+
+template <class TYPE>
+inline bool
+Test_Util<TYPE, false>::hasSameAllocator(const TYPE&, const TYPE&)
+{
+    return true;
+}
+
+template <class TYPE>
+bool checkAllocator(const TYPE& obj, const bsl::allocator<char>& allocator)
+    // check that the specified 'obj' uses the specified 'allocator'.
+{
+    return Test_Util<TYPE>::checkAllocator(obj, allocator);
+}
+
+template <class TYPE>
+bool hasSameAllocator(const TYPE& obj1, const TYPE& obj2)
+    // check that the specified 'obj1' and specified 'obj2' use the same
+    // allocator.
+{
+    return Test_Util<TYPE>::hasSameAllocator(obj1, obj2);
+}
+
+// ============================================================================
+//                          TEST DRIVER TEMPLATE
+// ----------------------------------------------------------------------------
+
+template <class TYPE>
+class TestDriver {
+    // This class template provides a namespace for testing the 'optional'
+    // type.
+  public:
+    static void bslstl_optional_test1();
```

Comment:
Please adjust to our usual naming.  Also, we typically have them in reverse order - highest case to lowest.

```suggestion
    static void testCase1();
```

Reply:
renamed and reordered